azurerm/internal/services/azurestackhci/parse/hci_cluster.go

```diff
@@ -0,0 +1,33 @@
+package parse
```

Comment:
Can we generate this file by creating a *resourceids.go* under the *azurestackhci* directory with following content:

```
package azurestackhci
 
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=HCIClusterId -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AzureStackHCI/Clusters/cluster1
```

Reply:
Could you share me an example PR? Thanks.

Reply:
https://github.com/magodo/terraform-provider-azurerm/blob/524170e685640a10101b6e0ac787ad06f8924cdc/azurerm/internal/services/network/resourceids.go#L2-L2

Reply:
updated

Reply:
I've submit a PR #9748 for this.

Reply:
To be honest this likely wants to be named `Cluster` since that's what this is - HCI is the product name so it's unnecessary in this context, but agreed we should update the generator to handle this case

Reply:
Once PR 9748 is merged, I will update this PR for this case.

Reply:
@neil-yechenwei this PR is not dependent on 9748, we can call this `Cluster` as mentioned above.

Reply:
updated. thx